pkg/utils/disruption/disruption.go

```diff
@@ -36,9 +36,9 @@ import (
 // disruption cost is highest, and it approaches zero as the node ages towards its expiration time.
 func LifetimeRemaining(clock clock.Clock, nodePool *v1.NodePool, nodeClaim *v1.NodeClaim) float64 {
 	remaining := 1.0
-	if nodePool.Spec.Disruption.ExpireAfter.Duration != nil {
+	if nodePool.Spec.Template.Spec.ExpireAfter.Duration != nil {
```

Comment:
Does it matter here where we are getting the expireAfter value? NodePool or NodeClaims?  Only asked for the case of a drifted NodeClaims where the two values can be different

Reply:
That's a good point, this should check the nodeclaim.